File: app/connections/actions.py
from .queries import GET_CONNECTIONS,ADD_CONNECTION,DELETE_CONNECTION
from app.utils import get_query_exec,save_query_exec,delete_query_exec,get_id_by_token
from app.models.connection import Connection
def get_connections(user_id):
    data={'user_id':user_id}
    rows = get_query_exec(GET_CONNECTIONS,data)
    if rows is None:
            return "Error: get connections query failed",400
    connections=[]
    for row in rows:
        id=row['id']
        user_id=row['user_id']
        name=row['name']
        position=row['position']
        company=row['company']
        phone=row['phone']
        accounts=row['accounts']
        connection = Connection(id,name,position,company,phone,accounts)
        connection= connection.__dict__
        connections.append(connection)  
    return connections,None

def add_connection(connection,user_id):
    #data={'user_id':data[0],'name':data[1],'position':data[2],'company':data[3],'phone':data[4],'accounts':data[5]}
    del connection.id
    connection = connection.__dict__
    user_id={'user_id': user_id}
    combined_dict = {**connection, **user_id}
    return save_query_exec(ADD_CONNECTION,combined_dict)


def delete_connection(user_id,connection_to_delete):
    data={"user_id":user_id,"id":connection_to_delete}
    delete_query_exec(DELETE_CONNECTION,data)

File: app/templates/js/connections/actions.js
import { sendGetRequest, getCookieValue,sendData,deleteData, getUserSession} from "../utils.js";
import { Connection } from "../models/connections.js";
import {createTableRow,createTable,createInsertForm,deleteButtonInit} from "../htmlUtils.js"

export async function initPage(){
    //Init variables
    var body = document.getElementById("table-wrap");
    let headers = ['Name', 'Position','Company', 'Phone', 'Accounts (Coming soon..)','Actions'];
    let inputTitles=['Name', 'Position','Company', 'Phone', 'Accounts'];
    //Create table
    var tbody = await createTable(body,headers);
    createInsertForm(inputTitles,"",addConnection);
    //Get data
    const connections = await getConnections();
    //Iterate
    for (let connection of connections) {
        var conn = new Connection(connection['id'],connection['name'],connection['position'],connection['company'],connection['phone'],connection['accountsData']);
        var connectionIdToDelete = conn.id;
        delete conn.id;
        createTableRow(tbody, conn);
        deleteButtonInit(tbody,deleteConnection,connectionIdToDelete);
       
    }
    document.getElementById("addRowButton").addEventListener('click', function() {
        addConnection(); 
    });
    
}

export async function getConnections() {
    const res = await sendGetRequest(`/connections/${getUserSession()}`);
    return res;
}

function deleteConnection(connectionIdToDelete) {
    var res = deleteData('delete-connection',connectionIdToDelete);
    if(res == false){
        console.error("Failed to delete the connection");
        return false;
    }
    return true;
};


async function addConnection() {
    // Retrieve input values
    var newName = document.getElementById("newName").value;
    var newPosition = document.getElementById("newPosition").value;
    var newCompany = document.getElementById("newCompany").value;
    var newPhone = document.getElementById("newPhone").value;
    var newAccounts = document.getElementById("newAccounts").value;
    // Create new Connection object
    var newConnection = new Connection("-1",newName, newPosition, newCompany, newPhone,newAccounts);
    delete newConnection.id;
    var token = getUserSession();
    var data=[token,newConnection]
    // Send data asynchronously
    var res = await sendData("/connections", newConnection); // Assuming sendData returns true/false
    // Check result
    if (res === false) {
        return false; // Handle error if sendData failed
    } else {
        var tbody=document.getElementById("table");
        var idConnection=res;
        delete newConnection._id;
        createTableRow(tbody,newConnection); // Call createNewRow if sendData was successful
        document.getElementById("newName").value="";
        document.getElementById("newPosition").value="";
        document.getElementById("newCompany").value="";
        document.getElementById("newPhone").value="";
        document.getElementById("newAccounts").value="";
        deleteButtonInit(tbody,deleteConnection,idConnection);
    }
    return true;
}
